src/workerd/api/http.c++

```diff
@@ -1092,12 +1092,24 @@ jsg::Ref<Response> Response::redirect(
 
   // TODO(conform): The URL is supposed to be parsed relative to the "current setting's object's API
   //   base URL".
-  auto urlOptions = kj::Url::Options { .percentDecode = false, .allowEmpty = true };
-  auto maybeParsedUrl = kj::Url::tryParse(url, kj::Url::REMOTE_HREF, urlOptions);
-  if (maybeParsedUrl == nullptr) {
-    JSG_FAIL_REQUIRE(TypeError, kj::str("Unable to parse URL: ", url));
+  kj::StringPtr parsedUrl = nullptr;
+  if (flags.getSpecCompliantUrlRedirect()) {
+    auto maybeParsedUrl = url::URL::parse(jsg::usv(url));
```

Comment:
@jasnell do you know if it would be better (and equivalent) to accept a `jsg::UsvString` as the `url` parameter to this function, then use it directly here (instead of `jsg::usv()`), and convert it to UTF-8 in the original code path, below?

Reply:
Yes.

Reply:
Done.